habitat-lab/habitat/tasks/rearrange/rearrange_sim.py

```diff
@@ -732,7 +721,90 @@ def _add_objs(
+                    info["translation"] = np.array(tar_translation)
+                    target_receptacles.extend(
+                        parse_receptacles_from_user_config(
+                            user_attr,
+                            parent_object_handle=tar_recep_handle,
+                            parent_template_directory=tar_path,
+                            mp3d=True,
+                            info=info
+                        )
+                    )
+                receptacles.extend(target_receptacles)
+                for i, (goal_recep_handle, goal_transform, goal_translation) in enumerate(ep_info.goal_receptacles):
+                    if goal_recep_handle.endswith('_:0000'):
+                        goal_recep_name = goal_recep_handle[:-6]
+                    goal_template = otm.get_templates_by_handle_substring(
+                        goal_recep_name
+                    )
+                    if not goal_template:
+                        raise ValueError(
+                            f"Template not found for object with handle {goal_recep_handle}"
+                        )
+                    goal_path = list(goal_template.keys())[0]
+                    goal_receptacle = rom.add_object_by_template_handle(
+                        goal_path
+                    )
+                    goal_receptacle.transformation = mn.Matrix4(
+                        [[goal_transform[j][i] for j in range(4)] for i in range(4)]
+                    )
+                    goal_receptacle.translation = goal_translation
+                    user_attr = goal_receptacle.user_attributes
+                    info["translation"] = np.array(goal_translation)
+                    goal_receptacles.extend(
+                        parse_receptacles_from_user_config(
+                            user_attr,
+                            parent_object_handle=goal_recep_handle,
+                            parent_template_directory=goal_path,
+                            mp3d=True,
+                            info=info
+                        )
+                    )
+                receptacles.extend(goal_receptacles)
+
+                receps = {}
+                for recep in receptacles:
+                    recep = cast(AABBReceptacle, recep)
+                    local_bounds = recep.bounds
+                    global_T = recep.get_global_transform(self)
+                    bounds = np.stack(
+                        [
+                            global_T.transform_point(local_bounds.min),
+                            global_T.transform_point(local_bounds.max),
+                        ],
+                        axis=0,
+                    )
+                    receps[recep.unique_name] = mn.Range3D(
+                        np.min(bounds, axis=0), np.max(bounds, axis=0)
+                    )
+                self._receptacles_cache[ep_info.scene_id] = receps
+                self._receptacles = receps
+    
```

Comment:
Can we wrap this snippet of code to a function in utils, for example? I don't want to dilute the code in core classes. 
If this requires a lot of effort, then simply ignore my comment here. :)

Reply:
Cross-reference will happen after wrapping this snippet into utils.py, I wrapped it into a class function instead.